lib/galaxy/celery/tasks.py

```diff
@@ -83,15 +83,15 @@ def export_history(
         app: MinimalManagerApp,
         sa_session: galaxy_scoped_session,
         job_manager: JobManager,
-        store_directory,
-        history_id,
-        job_id,
+        store_directory: str,
+        history_id: int,
+        job_id: int,
         include_hidden=False,
         include_deleted=False):
     history = sa_session.query(model.History).get(history_id)
     with model.store.DirectoryModelExportStore(store_directory, app=app, export_files="symlink") as export_store:
         export_store.export_history(history, include_hidden=include_hidden, include_deleted=include_deleted)
-    job = sa_session.query(model.Job).get(job_id)
+    job = sa_session.query(model.Job).filter_by(id=job_id).one()
```

Comment:
This is a minor thing, but I think that would then force a select even if the object is in the session. If we need to ensure an error is raised if no job is found, maybe raise it explicitly if the `get` returns `None`?

Reply:
Hmm, that is an interesting point: We should probably ensure tasks receive a fresh session, but I think the scoping mechanism we have now will just ensure that the process, thread or request get their own session.

Reply:
Yes, that's my understanding of how scoped_session works too. I think if we needed a separate session, we could use `session_factory` stored in the `scoped_session` to create one ([docs](https://docs.sqlalchemy.org/en/20/orm/contextual.html#sqlalchemy.orm.scoping.scoped_session.session_factory))